### PR TITLE
Support Java duration. Naming choices are up for discussion

### PR DESCRIPTION
**Pull request by rberenguel**
I've seen in the Scala source that jDuration might be more standard, but seems to be “less classy”). Also, since both Durations have the same SimpleName had to change the namer in the test suite, not sure if this is the best choice. I removed the limitation on the generated duration, though, java.time.Duration seems to handle that specific number just fine

### CODE REVIEW

**Comment by rberenguel** [comment]
Why is coveralls taking ages to report back?

File: core/src/test/scala/pureconfig/ConfigConvertChecks.scala
Comment (melrief) [inline core/src/test/scala/pureconfig/ConfigConvertChecks.scala:81]: I'm ok with this change if everybody else is ok

**Review by melrief — APPROVED** [review]
Good job @rberenguel! The code looks follows very well the repository standard and the tests are well done. Thank you very much. I'll wait for another review and then merge.

**Review by derekmorr — APPROVED** [review]

File: core/src/test/scala/pureconfig/ConfigConvertChecks.scala
Comment (ruippeixotog) [reply core/src/test/scala/pureconfig/ConfigConvertChecks.scala:81]: I'm ok with it, in this case I suppose it helps to distinguish a Scala Duration from a Java one.

**Review by ruippeixotog — APPROVED** [review]
LGTM, nice job!

**Comment by rberenguel** [comment]
Thanks @ruippeixotog @derekmorr @melrief :)

**Comment by ruippeixotog** [comment]
@rberenguel, there seems to be one problem with this PR, unfortunately :( In my computer I'm having some non-deterministic errors in the conversion of `java.time.Duration`:

```
[info] - should read an arbitrary java.time.Duration *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: PT-1525073H-41M0.327123929S did not equal PT-1525073H-41M-0.327123929S
[info]     Location: (ConfigConvertChecks.scala:31)
[info]     Occurred when passed generated values (
[info]       arg0 = PT-1525073H-41M-0.327123929S
[info]     )
```

```
[info] - should read an arbitrary java.time.Duration *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: PT-256047H-2M0.448787857S did not equal PT-256047H-2M-0.448787857S
[info]     Location: (ConfigConvertChecks.scala:31)
[info]     Occurred when passed generated values (
[info]       arg0 = PT-256047H-2M-0.448787857S
[info]     )
```

Do you know what might that be?

**Comment by ruippeixotog** [comment]
Just happened in [Travis, too](https://travis-ci.org/pureconfig/pureconfig/jobs/230512293).

**Comment by rberenguel** [comment]
@ruippeixotog yikes! I ran it three times, probabilistic testing ftw :( I'll check right after dinner, curious to see what the issue might be

In all three cases the smallest sign is reversed, smells of overflow...

**Pull request merged**
